### PR TITLE
フロントエンドとバックエンドのコンテナを一度に管理できるように変更

### LINKED ISSUE #90: dockerの起動を簡潔にするためにdocker-composeを一つにまとめる
## ✨ Summary

今までは`frontend/`と`backend/`のそれぞれに`docker-compose.yml`を置いていたが、コンテナ作成時にディレクトリを移動する必要があり手間だったため、`docker-compose.yml`を一つにまとめて使いやすくする。
## 🔥 Requirements

- `frontend/`と`backend/`のそれぞれにあるdocker-compose.ymlを一つにまとめて管理できるようにする
- 2つのコンテナの片方もしくは両方をまとめて`起動`できるように`Makefile`を作成する
- 2つのコンテナの片方もしくは両方をまとめて`停止`ができるように`Makefile`を編集する
- コンテナを指定して`ターミナルにアクセス`できるように`Makefile`を編集する

## ✅ Tasks

- [x] `docker-compose.ym`lの作成
- [x] `Makefile`の作成
- [x] ドキュメントを変更に対応させる

### PR DESCRIPTION
**Pull request by mitukawa**
## 🔨 Details of Changes

- リポジトリの`root`でバックエンドとフロントエンドのDockerコンテナを管理できるようにした。
- コンテナの起動や停止、ターミナルへのアクセスといった実行を行うコマンドをまとめたMakefileを作成した。
- Makefileから`linter/formater`やFlaskの開発サーバーを起動できるようにした。

## ✅ Resolved Issues

- close #90

## 🤝 Related Issues

- #90

### CODE REVIEW

**Comment by watagit** [comment]
`make up`起動時にこのようなエラーが出てしまいます 🤔 

```
 => ERROR [sake_backend  5/11] RUN apt-get install -y git                                                                                                                                                      4.7s
------
 > [sake_backend  5/11] RUN apt-get install -y git:
#17 0.257 Reading package lists...
#17 1.333 Building dependency tree...
#17 1.505 Reading state information...
#17 1.676 The following additional packages will be installed:
#17 1.676   git-man libbsd0 libcurl3-gnutls libedit2 liberror-perl libssl1.0.0 libx11-6
#17 1.676   libx11-data libxau6 libxcb1 libxdmcp6 libxext6 libxmuu1 openssh-client xauth
#17 1.678 Suggested packages:
#17 1.678   gettext-base git-daemon-run | git-daemon-sysvinit git-doc git-el git-email
#17 1.678   git-gui gitk gitweb git-cvs git-mediawiki git-svn keychain libpam-ssh
#17 1.678   monkeysphere ssh-askpass
#17 1.744 The following NEW packages will be installed:
#17 1.744   git git-man libbsd0 libcurl3-gnutls libedit2 liberror-perl libssl1.0.0
#17 1.745   libx11-6 libx11-data libxau6 libxcb1 libxdmcp6 libxext6 libxmuu1
#17 1.745   openssh-client xauth
#17 2.327 0 upgraded, 16 newly installed, 0 to remove and 9 not upgraded.
#17 2.327 Need to get 7593 kB of archives.
#17 2.327 After this operation, 46.1 MB of additional disk space will be used.
#17 2.327 Get:1 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libxau6 amd64 1:1.0.8-1ubuntu1 [7556 B]
#17 2.594 Get:2 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libbsd0 amd64 0.8.7-1ubuntu0.1 [41.6 kB]
#17 2.857 Get:3 http://archive.ubuntu.com/ubuntu bionic/main amd64 libxdmcp6 amd64 1:1.1.2-3 [10.7 kB]
#17 2.903 Get:4 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libxcb1 amd64 1.13-2~ubuntu18.04 [45.5 kB]
#17 3.166 Get:5 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libx11-data all 2:1.6.4-3ubuntu0.4 [114 kB]
#17 3.468 Get:6 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libx11-6 amd64 2:1.6.4-3ubuntu0.4 [572 kB]
#17 3.974 Get:7 http://archive.ubuntu.com/ubuntu bionic/main amd64 libxext6 amd64 2:1.3.3-1 [29.4 kB]
#17 3.986 Get:8 http://archive.ubuntu.com/ubuntu bionic/main amd64 libedit2 amd64 3.1-20170329-1 [76.9 kB]
#17 4.056 Ign:9 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libssl1.0.0 amd64 1.0.2n-1ubuntu5.6
#17 4.067 Get:10 http://archive.ubuntu.com/ubuntu bionic/main amd64 libxmuu1 amd64 2:1.1.2-2 [9674 B]
#17 4.067 Get:11 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 openssh-client amd64 1:7.6p1-4ubuntu0.5 [612 kB]
#17 4.267 Get:12 http://archive.ubuntu.com/ubuntu bionic/main amd64 xauth amd64 1:1.0.10-1 [24.6 kB]
#17 4.272 Ign:13 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libcurl3-gnutls amd64 7.58.0-2ubuntu3.14
#17 4.272 Get:14 http://archive.ubuntu.com/ubuntu bionic/main amd64 liberror-perl all 0.17025-1 [22.8 kB]
#17 4.277 Ign:15 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 git-man all 1:2.17.1-1ubuntu0.8
#17 4.277 Ign:16 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 git amd64 1:2.17.1-1ubuntu0.8
#17 4.494 Err:9 http://security.ubuntu.com/ubuntu bionic-updates/main amd64 libssl1.0.0 amd64 1.0.2n-1ubuntu5.6
#17 4.494   404  Not Found [IP: 91.189.88.152 80]
#17 4.688 Err:13 http://security.ubuntu.com/ubuntu bionic-updates/main amd64 libcurl3-gnutls amd64 7.58.0-2ubuntu3.14
#17 4.688   404  Not Found [IP: 91.189.88.152 80]
#17 4.688 Err:15 http://security.ubuntu.com/ubuntu bionic-updates/main amd64 git-man all 1:2.17.1-1ubuntu0.8
#17 4.688   404  Not Found [IP: 91.189.88.152 80]
#17 4.688 Err:16 http://security.ubuntu.com/ubuntu bionic-updates/main amd64 git amd64 1:2.17.1-1ubuntu0.8
#17 4.688   404  Not Found [IP: 91.189.88.152 80]
#17 4.700 Fetched 1566 kB in 3s (535 kB/s)
#17 4.700 E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/o/openssl1.0/libssl1.0.0_1.0.2n-1ubuntu5.6_amd64.deb  404  Not Found [IP: 91.189.88.152 80]
#17 4.701 E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/c/curl/libcurl3-gnutls_7.58.0-2ubuntu3.14_amd64.deb  404  Not Found [IP: 91.189.88.152 80]
#17 4.701 E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/g/git/git-man_2.17.1-1ubuntu0.8_all.deb  404  Not Found [IP: 91.189.88.152 80]
#17 4.703 E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/g/git/git_2.17.1-1ubuntu0.8_amd64.deb  404  Not Found [IP: 91.189.88.152 80]
#17 4.703 E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
------
failed to solve: rpc error: code = Unknown desc = executor failed running [/bin/bash -c apt-get install -y git]: exit code: 100
```

**Comment by mitukawa** [comment]
自分の環境では動いたけど、gitのインストールは念のため消しときます。

> `make up`実行時にこのようなエラーが出てしまいます 🤔
> 
> ```
>  => ERROR [sake_backend  5/11] RUN apt-get install -y git                                                                                                                                                      4.7s
> ------
>  > [sake_backend  5/11] RUN apt-get install -y git:
> #17 0.257 Reading package lists...
> #17 1.333 Building dependency tree...
> #17 1.505 Reading state information...
> #17 1.676 The following additional packages will be installed:
> #17 1.676   git-man libbsd0 libcurl3-gnutls libedit2 liberror-perl libssl1.0.0 libx11-6
> #17 1.676   libx11-data libxau6 libxcb1 libxdmcp6 libxext6 libxmuu1 openssh-client xauth
> #17 1.678 Suggested packages:
> #17 1.678   gettext-base git-daemon-run | git-daemon-sysvinit git-doc git-el git-email
> #17 1.678   git-gui gitk gitweb git-cvs git-mediawiki git-svn keychain libpam-ssh
> #17 1.678   monkeysphere ssh-askpass
> #17 1.744 The following NEW packages will be installed:
> #17 1.744   git git-man libbsd0 libcurl3-gnutls libedit2 liberror-perl libssl1.0.0
> #17 1.745   libx11-6 libx11-data libxau6 libxcb1 libxdmcp6 libxext6 libxmuu1
> #17 1.745   openssh-client xauth
> #17 2.327 0 upgraded, 16 newly installed, 0 to remove and 9 not upgraded.
> #17 2.327 Need to get 7593 kB of archives.
> #17 2.327 After this operation, 46.1 MB of additional disk space will be used.
> #17 2.327 Get:1 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libxau6 amd64 1:1.0.8-1ubuntu1 [7556 B]
> #17 2.594 Get:2 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libbsd0 amd64 0.8.7-1ubuntu0.1 [41.6 kB]
> #17 2.857 Get:3 http://archive.ubuntu.com/ubuntu bionic/main amd64 libxdmcp6 amd64 1:1.1.2-3 [10.7 kB]
> #17 2.903 Get:4 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libxcb1 amd64 1.13-2~ubuntu18.04 [45.5 kB]
> #17 3.166 Get:5 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libx11-data all 2:1.6.4-3ubuntu0.4 [114 kB]
> #17 3.468 Get:6 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libx11-6 amd64 2:1.6.4-3ubuntu0.4 [572 kB]
> #17 3.974 Get:7 http://archive.ubuntu.com/ubuntu bionic/main amd64 libxext6 amd64 2:1.3.3-1 [29.4 kB]
> #17 3.986 Get:8 http://archive.ubuntu.com/ubuntu bionic/main amd64 libedit2 amd64 3.1-20170329-1 [76.9 kB]
> #17 4.056 Ign:9 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libssl1.0.0 amd64 1.0.2n-1ubuntu5.6
> #17 4.067 Get:10 http://archive.ubuntu.com/ubuntu bionic/main amd64 libxmuu1 amd64 2:1.1.2-2 [9674 B]
> #17 4.067 Get:11 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 openssh-client amd64 1:7.6p1-4ubuntu0.5 [612 kB]
> #17 4.267 Get:12 http://archive.ubuntu.com/ubuntu bionic/main amd64 xauth amd64 1:1.0.10-1 [24.6 kB]
> #17 4.272 Ign:13 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libcurl3-gnutls amd64 7.58.0-2ubuntu3.14
> #17 4.272 Get:14 http://archive.ubuntu.com/ubuntu bionic/main amd64 liberror-perl all 0.17025-1 [22.8 kB]
> #17 4.277 Ign:15 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 git-man all 1:2.17.1-1ubuntu0.8
> #17 4.277 Ign:16 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 git amd64 1:2.17.1-1ubuntu0.8
> #17 4.494 Err:9 http://security.ubuntu.com/ubuntu bionic-updates/main amd64 libssl1.0.0 amd64 1.0.2n-1ubuntu5.6
> #17 4.494   404  Not Found [IP: 91.189.88.152 80]
> #17 4.688 Err:13 http://security.ubuntu.com/ubuntu bionic-updates/main amd64 libcurl3-gnutls amd64 7.58.0-2ubuntu3.14
> #17 4.688   404  Not Found [IP: 91.189.88.152 80]
> #17 4.688 Err:15 http://security.ubuntu.com/ubuntu bionic-updates/main amd64 git-man all 1:2.17.1-1ubuntu0.8
> #17 4.688   404  Not Found [IP: 91.189.88.152 80]
> #17 4.688 Err:16 http://security.ubuntu.com/ubuntu bionic-updates/main amd64 git amd64 1:2.17.1-1ubuntu0.8
> #17 4.688   404  Not Found [IP: 91.189.88.152 80]
> #17 4.700 Fetched 1566 kB in 3s (535 kB/s)
> #17 4.700 E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/o/openssl1.0/libssl1.0.0_1.0.2n-1ubuntu5.6_amd64.deb  404  Not Found [IP: 91.189.88.152 80]
> #17 4.701 E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/c/curl/libcurl3-gnutls_7.58.0-2ubuntu3.14_amd64.deb  404  Not Found [IP: 91.189.88.152 80]
> #17 4.701 E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/g/git/git-man_2.17.1-1ubuntu0.8_all.deb  404  Not Found [IP: 91.189.88.152 80]
> #17 4.703 E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/g/git/git_2.17.1-1ubuntu0.8_amd64.deb  404  Not Found [IP: 91.189.88.152 80]
> #17 4.703 E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
> ------
> failed to solve: rpc error: code = Unknown desc = executor failed running [/bin/bash -c apt-get install -y git]: exit code: 100
> ```
自分の環境では問題なく動いたけど、念のためgitのインストールは消しときます。

> `make up`実行時にこのようなエラーが出てしまいます 🤔
> 
> ```
>  => ERROR [sake_backend  5/11] RUN apt-get install -y git                                                                                                                                                      4.7s
> ------
>  > [sake_backend  5/11] RUN apt-get install -y git:
> #17 0.257 Reading package lists...
> #17 1.333 Building dependency tree...
> #17 1.505 Reading state information...
> #17 1.676 The following additional packages will be installed:
> #17 1.676   git-man libbsd0 libcurl3-gnutls libedit2 liberror-perl libssl1.0.0 libx11-6
> #17 1.676   libx11-data libxau6 libxcb1 libxdmcp6 libxext6 libxmuu1 openssh-client xauth
> #17 1.678 Suggested packages:
> #17 1.678   gettext-base git-daemon-run | git-daemon-sysvinit git-doc git-el git-email
> #17 1.678   git-gui gitk gitweb git-cvs git-mediawiki git-svn keychain libpam-ssh
> #17 1.678   monkeysphere ssh-askpass
> #17 1.744 The following NEW packages will be installed:
> #17 1.744   git git-man libbsd0 libcurl3-gnutls libedit2 liberror-perl libssl1.0.0
> #17 1.745   libx11-6 libx11-data libxau6 libxcb1 libxdmcp6 libxext6 libxmuu1
> #17 1.745   openssh-client xauth
> #17 2.327 0 upgraded, 16 newly installed, 0 to remove and 9 not upgraded.
> #17 2.327 Need to get 7593 kB of archives.
> #17 2.327 After this operation, 46.1 MB of additional disk space will be used.
> #17 2.327 Get:1 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libxau6 amd64 1:1.0.8-1ubuntu1 [7556 B]
> #17 2.594 Get:2 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libbsd0 amd64 0.8.7-1ubuntu0.1 [41.6 kB]
> #17 2.857 Get:3 http://archive.ubuntu.com/ubuntu bionic/main amd64 libxdmcp6 amd64 1:1.1.2-3 [10.7 kB]
> #17 2.903 Get:4 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libxcb1 amd64 1.13-2~ubuntu18.04 [45.5 kB]
> #17 3.166 Get:5 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libx11-data all 2:1.6.4-3ubuntu0.4 [114 kB]
> #17 3.468 Get:6 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libx11-6 amd64 2:1.6.4-3ubuntu0.4 [572 kB]
> #17 3.974 Get:7 http://archive.ubuntu.com/ubuntu bionic/main amd64 libxext6 amd64 2:1.3.3-1 [29.4 kB]
> #17 3.986 Get:8 http://archive.ubuntu.com/ubuntu bionic/main amd64 libedit2 amd64 3.1-20170329-1 [76.9 kB]
> #17 4.056 Ign:9 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libssl1.0.0 amd64 1.0.2n-1ubuntu5.6
> #17 4.067 Get:10 http://archive.ubuntu.com/ubuntu bionic/main amd64 libxmuu1 amd64 2:1.1.2-2 [9674 B]
> #17 4.067 Get:11 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 openssh-client amd64 1:7.6p1-4ubuntu0.5 [612 kB]
> #17 4.267 Get:12 http://archive.ubuntu.com/ubuntu bionic/main amd64 xauth amd64 1:1.0.10-1 [24.6 kB]
> #17 4.272 Ign:13 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libcurl3-gnutls amd64 7.58.0-2ubuntu3.14
> #17 4.272 Get:14 http://archive.ubuntu.com/ubuntu bionic/main amd64 liberror-perl all 0.17025-1 [22.8 kB]
> #17 4.277 Ign:15 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 git-man all 1:2.17.1-1ubuntu0.8
> #17 4.277 Ign:16 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 git amd64 1:2.17.1-1ubuntu0.8
> #17 4.494 Err:9 http://security.ubuntu.com/ubuntu bionic-updates/main amd64 libssl1.0.0 amd64 1.0.2n-1ubuntu5.6
> #17 4.494   404  Not Found [IP: 91.189.88.152 80]
> #17 4.688 Err:13 http://security.ubuntu.com/ubuntu bionic-updates/main amd64 libcurl3-gnutls amd64 7.58.0-2ubuntu3.14
> #17 4.688   404  Not Found [IP: 91.189.88.152 80]
> #17 4.688 Err:15 http://security.ubuntu.com/ubuntu bionic-updates/main amd64 git-man all 1:2.17.1-1ubuntu0.8
> #17 4.688   404  Not Found [IP: 91.189.88.152 80]
> #17 4.688 Err:16 http://security.ubuntu.com/ubuntu bionic-updates/main amd64 git amd64 1:2.17.1-1ubuntu0.8
> #17 4.688   404  Not Found [IP: 91.189.88.152 80]
> #17 4.700 Fetched 1566 kB in 3s (535 kB/s)
> #17 4.700 E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/o/openssl1.0/libssl1.0.0_1.0.2n-1ubuntu5.6_amd64.deb  404  Not Found [IP: 91.189.88.152 80]
> #17 4.701 E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/c/curl/libcurl3-gnutls_7.58.0-2ubuntu3.14_amd64.deb  404  Not Found [IP: 91.189.88.152 80]
> #17 4.701 E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/g/git/git-man_2.17.1-1ubuntu0.8_all.deb  404  Not Found [IP: 91.189.88.152 80]
> #17 4.703 E: Failed to fetch http://security.ubuntu.com/ubuntu/pool/main/g/git/git_2.17.1-1ubuntu0.8_amd64.deb  404  Not Found [IP: 91.189.88.152 80]
> #17 4.703 E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
> ------
> failed to solve: rpc error: code = Unknown desc = executor failed running [/bin/bash -c apt-get install -y git]: exit code: 100
> ```

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+sh-frontend:
+	docker-compose exec frontend bash
+
+sh-backend:
+	docker-compose exec backend bash
+
+
+# 起動中のコンテナの確認
+view-run-container:
+	docker ps
```
Comment (watagit) [inline Makefile:31]: この部分についてですが、`make view-run-container`と入力するより`docker ps`と打ったほうが短いので、`make ps`とかにしても良いかもしれないです 🤔

File: README.md [hunk 2]
```diff
@@ -1,3 +1,22 @@
...
+- `make sh-{backend | frontend}`
+makeが実行出来ない場合
+- `docker-compose exec {backend | frontend} bash`
+### 開発サーバーの起動  
+- `make run-flask-app`  
+他にも`Makefile`に頻繁に使用するコマンドをまとめたので必要があれば参考にしてください
+
+
+
+
```
Comment (watagit) [inline README.md:22]: [nits]
フォルダ末尾の空行は1行でお願いします〜！

File: Makefile (hunk 1 above)
Comment (mitukawa) [reply Makefile:31]: 確かに、長かった意味ないね
`make ps`にします。

**Comment by mitukawa** [comment]
割り方との表記を合わせるなら
`アスパラ`や`生ハム`なども太字にした方がいいと思うけど、どうですか？

---
image

**Comment by mitukawa** [comment]
このPRでは、レイアウトは考慮しないとのことだったので、大丈夫です。

**Review by watagit — APPROVED** [review]
[LGTM](https://www.lgtm.app/i/XvqHgCQMG)

**Comment by mitukawa** [comment]
あざす

**Pull request merged**